sound/soc/soc-compress.c

```diff
@@ -77,9 +78,16 @@ static int soc_compr_open(struct snd_compr_stream *cstream)
 {
 	struct snd_soc_pcm_runtime *rtd = cstream->private_data;
 	struct snd_soc_component *component;
+	struct snd_soc_rtdcom_list *rtdcom;
 	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
 	int ret;
 
+	for_each_rtdcom(rtd, rtdcom) {
+		component = rtdcom->component;
+
+		pm_runtime_get_sync(component->dev);
```

Comment:
need to test for -EACCESS if pm_runtime is not enabled.

e.g.

````c

ret = pm_runtime_get_sync(slave->bus->dev);
	if (ret < 0 && ret != -EACCES) {
		pm_runtime_put_noidle(component->dev);
		return ret;
	}
````

Reply:
This code mimics soc-pcm behavior. soc_pcm_open does not test for access. Maybe I'm missing something here.

Reply:
if get_sync fails there is a reference leak. it's not because others do things wrong that we should do so as well.

Reply:
Ack.

Reply:
still open